src/System.Data.Common/src/System/Data/Common/IDbColumnSchemaGenerator.cs

```diff
@@ -0,0 +1,11 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Data.Common
+{
+    public interface IDbColumnSchemaGenerator
```

Comment:
The idea behind this interface is to develop/make the schema of the columns as a List of DbColumn. This may be done using the metadata that the providers may have with the data that they have in DbDataReader. 
The term "generate" would suitably describe the function of the classes implementing the interface.